static/app/views/dashboards/widgetBuilder/hooks/useWidgetBuilderState.tsx

```diff
@@ -315,4 +316,15 @@ function deserializeLimit(value: string): number {
   return decodeInteger(value, DEFAULT_RESULTS_LIMIT);
 }
 
+/**
+ * Decodes the query from the query params
+ * Returns an array with an empty string if the query is empty
+ */
+function deserializeQuery(queries: string[]): string[] {
+  if (queries.length === 0) {
+    return [''];
+  }
+  return queries;
+}
+
```

Comment:
this fixes the filters not working by ensuring there it as least one query element